server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -555,7 +566,9 @@ private static Tuple<Map<String, DataStream>, Map<String, DataStreamAlias>> getD
         List<String> requestedDataStreams = filterIndices(
             snapshotInfo.dataStreams(),
             Stream.of(requestIndices, featureStateDataStreams).flatMap(Collection::stream).toArray(String[]::new),
-            IndicesOptions.fromOptions(true, true, true, true)
+            DataStream.isFailureStoreFeatureFlagEnabled()
+                ? IndicesOptions.lenientExpandIncludeFailureStore()
```

Comment:
Looking at the filterIndices function - it doesn't seem to actually make any use of whether to include failure stores or not. Could we leave this unchanged?

Reply:
Good point, I made it more explicit in the javadoc and only kept the `IndicesOptions.lenientExpand()`.